api/filter/fault.proto

```diff
@@ -0,0 +1,124 @@
+syntax = "proto3";
+
+package envoy.api.v2.filter;
+
+import "api/rds.proto";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/wrappers.proto";
+
+// Delay specification is used to inject latency into the
+// HTTP/gRPC/Mongo/Redis operation or delay proxying of TCP connections.
+message FaultDelay {
+  enum FaultDelayType {
+    // Fixed delay (step function).
+    FIXED = 0;
+  }
+
+  // Delay type to use (fixed|exponential|..). Currently, only fixed delay (step function) is supported.
+  FaultDelayType type = 1;
+
+  // An integer between 0-100 indicating the percentage of operations/connection requests
+  // on which the delay will be injected.
```

Comment:
Maybe specify the default value if the field is not present?

Reply:
Thats the problem.. All these fields are required.. Unless we use istio style REQUIRED words everywhere, I am merely copying these from the envoy docs.

Reply:
The wrapped primitive type is a signal that the field is optional. If it's always required, then you can use `uint32`.

Reply:
More like, I have to use uint32..